projects/core/src/cart/facade/selective-cart.service.ts

```diff
@@ -36,12 +38,21 @@ export class SelectiveCartService {
     protected store: Store<StateWithMultiCart>,
     protected userService: UserService,
     protected authService: AuthService,
-    protected multiCartService: MultiCartService
+    protected multiCartService: MultiCartService,
+    protected baseSiteService: BaseSiteService
   ) {
+    if (this.baseSiteService) {
+      this.baseSiteService
+        .getActive()
+        .subscribe(value => (this.activeBaseSite = value));
+    }
+
     this.userService.get().subscribe(user => {
```

Comment:
Could we `combineLatest(this.userService.get(), this.baseSiteService.getActive())` and work on this instead?
Then `activeBaseSite` property won't be needed.

Reply:
And we would react correctly on base site change which could not trigger `this.userService.get()` to emit again.